app/models/invitation.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class Invitation < ApplicationRecord
+  belongs_to :school
+  validates :email_address,
+            format: { with: EmailValidator.regexp, message: I18n.t('validations.invitation.email_address') }
+  validate :school_is_verified
+  after_create :send_invitation_email
+
+  generates_token_for :teacher_invitation, expires_in: 30.days do
+    email_address
```

Comment:
Minor: It might be nice to add a unit test that demonstrates what the call to `Invitation#email_address` in the block is meant to achieve. As I understand it, it should mean the token becomes invalid if it's verified against an invitation record with a different `email_address` than when the token was created.